src/mapping/bd_rhapsody/script.py

```diff
@@ -0,0 +1,407 @@
+import os
+import re
+import subprocess
+import tempfile
+import logging
+from sys import stdout
+from typing import Any
+import pandas as pd
+import shutil
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+console_handler = logging.StreamHandler(stdout)
+logFormatter = logging.Formatter("%(asctime)s %(levelname)-8s %(message)s")
+console_handler.setFormatter(logFormatter)
+logger.addHandler(console_handler)
+
+## VIASH START
+par = {
+  'input': ['resources_test/bdrhap_5kjrt/raw/12ABC_S1_L432_R1_001_subset.fastq.gz',
+            'resources_test/bdrhap_5kjrt/raw/12ABC_S1_L432_R2_001_subset.fastq.gz'],
+  'mode': 'wta',
+  'output': 'foo',
+  'subsample': None,
+  'reference': 'resources_test/bdrhap_ref_gencodev40_chr1/GRCh38_primary_assembly_genome_chr1.tar.gz',
+  'transcriptome_annotation': 'resources_test/bdrhap_ref_gencodev40_chr1/gencode_v40_annotation_chr1.gtf',
+  'exact_cell_count': None,
+  'disable_putative_calling': False,
+  'parallel': True,
+  'timestamps': False,
+  'abseq_reference': ["resources_test/bdrhap_5kjrt/raw/BDAbSeq_ImmuneDiscoveryPanel.fasta"],
+  'supplemental_reference': []
+}
+meta = {
+  'resources_dir': 'src/mapping/bd_rhapsody_wta',
+  'temp_dir': os.getenv("VIASH_TEMP")
+}
+## VIASH END
+
+def strip_margin(text: str) -> str:
```

Comment:
I think you can https://docs.python.org/3/library/textwrap.html#textwrap.dedent for this?

Reply:
No, because dedent removes too many spaces:
```python
import re
from textwrap import dedent

def strip_margin(text: str) -> str:
  return re.sub('\n[ \t]*\|', '\n', text)

print(strip_margin(f"""
  | - class: File
  |   location: foo.h5mu
  |"""))

print(dedent(f"""
   - class: File
     location: foo.h5mu
  """))
```

`strip_margin` produces:
```
 - class: File
   location: foo.h5mu
```

`dedent` produces:
```
- class: File
  location: foo.h5mu
```